engine/classes/ElggInstaller.php

```diff
@@ -113,6 +114,11 @@ protected function getApp() {
 			$config->elgg_config_locks = false;
 			$config->installer_running = true;
 			$config->dbencoding = 'utf8mb4';
+			$config->boot_cache_ttl = 0;
+			$config->system_cache_enabled = false;
+			$config->simplecache_enabled = false;
+			$config->debug = \Psr\Log\LogLevel::WARNING;
+			$config->cacheroot = Paths::project() . 'install/cache/';
```

Comment:
What if project path isn't writeable (as it should be)? Wouldn't `sys_get_temp_dir()` be better?

Reply:
I am not sure what `sys_get_temp_dir` is supposed to do. Last time I tried to use TempFile class you added, it didn't work for me.

Reply:
https://secure.php.net/manual/en/function.sys-get-temp-dir.php

Reply:
@hypeJunction i think the problem @jeabakker describes is valid. Can you agree with the suggestion?

Reply:
I am not going to spend any more time on this. Without setting cacheroot it just writes the cache into root anyway.

Reply:
> Can you agree with the suggestion?
As I said, last time I tried to use `TempFile` that uses this approach it didn't work, so I am not sure if it's a valid approach. But feel free to fork and do as you please.

Reply:
project root is not writeable as we suggest for security reasons, so this PR will cause issues (as does the current implementation).

Reply:
> last time I tried to use TempFile that uses this approach it didn't work

i am unaware of a problem with this class... could you create a ticket for it?

Reply:
Whatever